### DISCUSSION TITLE
Does SAM use Non-Manifold Topology?

### DISCUSSION THREAD

**Discussion by jpstaub** [HoareLea/SAM#284] [Q&A]
Hello,

I have been chasing down a better way to abstract analytical models from BIM for the purposes of: (1) loads analysis, (2) performance analysis.

Recently I became aware of [Topologic](https://topologic.app/) which uses Non-Manifold Topology (NMT) to produce analytical models. Based on documentation in several papers NMT looks to offer a better workflow relative to what I've been doing thus far in the Revit environment to abstract analytical models from BIM.

Further research on NMT workflow led me to SAM which also looks quite promising. However, from existing documentation I can't tell whether SAM uses NMT or if it adapts more common Manifold Topology (MT) to generate analytical models.

If you'd be so kind, please comment on whether SAM is NMT or MT based.

Best wishes,
Jake Staub

**Comment by michaldengusiak** [comment]
Hi Jake,

can you send us some examples in Rino... so we can test and see... we have few option to generate model including Topologic so best is to test

**Reply by jpstaub** [reply]
Hi Michal,

Thanks for your response.

I have not yet incorporated Rhino into my workflow. So I do not have Rhino examples to send.

I am just now starting to work with Dynamo/Topologic to generate models in Revit using backgrounds provided by architects as a basis. Based on your response it seems like SAM functions in a similar manner only using Rhino in combination with Topologic instead of Dynamo.

If you wouldn't mind sharing, what do you feel are the advantages of working in Rhino to generate models rather than a BIM authoring tool like Revit?

Best wishes,
Jake
